<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>FOXO3 Project</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">foxo</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">FOXO3 Project</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-02-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>foxo/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommparillafoxotree75ea1842548dec5d7fd65d5d15ec8025121f4e11targetblank75ea184a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mparilla/foxo/tree/75ea1842548dec5d7fd65d5d15ec8025121f4e11" target="_blank">75ea184</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommparillafoxotree75ea1842548dec5d7fd65d5d15ec8025121f4e11targetblank75ea184a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/GSE14879/
    Ignored:    data/GSE19069/
    Ignored:    data/GSE6338/
    Ignored:    output/.DS_Store
    Ignored:    output/combined_exprs.rds
    Ignored:    output/tcl_eset.rds

Untracked files:
    Untracked:  analysis/correlation.Rmd
    Untracked:  analysis/fgsea.Rmd
    Untracked:  analysis/foxo_ptcl_dge.Rmd
    Untracked:  analysis/pathway_enrichment.Rmd
    Untracked:  code/grn.R
    Untracked:  es_10k/
    Untracked:  output/annotLookup.rds
    Untracked:  output/differential_gene_expression_ptcl_vs_control.xlsx
    Untracked:  output/mart.rds
    Untracked:  output/paper_figures/
    Untracked:  output/tcl_es.rds
    Untracked:  output/tcl_eset_foxo3_active.rds
    Untracked:  output/toptable_foxoactivevsnon.csv
    Untracked:  output/toptable_ptcl.vs.control.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mparilla/foxo/blob/75ea1842548dec5d7fd65d5d15ec8025121f4e11/analysis/index.Rmd" target="_blank">75ea184</a>
</td>
<td>
mleukam
</td>
<td>
2020-02-02
</td>
<td>
wflow_git_commit(all = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mparilla/foxo/042feba9558aa5b05dce92a1d859141e9155fd2d/docs/index.html" target="_blank">042feba</a>
</td>
<td>
mleukam
</td>
<td>
2019-09-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mparilla/foxo/blob/0614c4ed142f006d5fd0ddbe45a2f4f30653c248/analysis/index.Rmd" target="_blank">0614c4e</a>
</td>
<td>
mleukam
</td>
<td>
2019-09-21
</td>
<td>
added link to dge notebook
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mparilla/foxo/fdd213b997e8e89d010e59b5abe2719d7ab26500/docs/index.html" target="_blank">fdd213b</a>
</td>
<td>
mleukam
</td>
<td>
2019-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mparilla/foxo/blob/7452f995e35935d6137c25f56bf1e91bbdbb1568/analysis/index.Rmd" target="_blank">7452f99</a>
</td>
<td>
mleukam
</td>
<td>
2019-09-17
</td>
<td>
initial website build
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mparilla/foxo/fdf12acb594406f9e0d153b37730b39c9f930627/docs/index.html" target="_blank">fdf12ac</a>
</td>
<td>
mleukam
</td>
<td>
2019-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mparilla/foxo/90b885d3939d307d924acb47f1d0b297159962c2/docs/index.html" target="_blank">90b885d</a>
</td>
<td>
mleukam
</td>
<td>
2019-09-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mparilla/foxo/blob/11ae6ed31edcb9bcc14e4811d3e572b4cb798802/analysis/index.Rmd" target="_blank">11ae6ed</a>
</td>
<td>
mleukam
</td>
<td>
2019-09-17
</td>
<td>
initial website build
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mparilla/foxo/blob/c0e972d4e7e8c502e8a66d9faa77904ec75408d9/analysis/index.Rmd" target="_blank">c0e972d</a>
</td>
<td>
mleukam
</td>
<td>
2019-09-12
</td>
<td>
gene sets and updated notebooks
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mparilla/foxo/blob/d9ba9732847f2df1ec3f001f5a56183a97da6701/analysis/index.Rmd" target="_blank">d9ba973</a>
</td>
<td>
mleukam
</td>
<td>
2019-09-12
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Welcome to the computational lab notebook for a project examining the role of FOXO3 in T-cell lymphoma.</p>
<div id="data-sources" class="section level2">
<h2>Data Sources</h2>
<p>Downloaded raw data from GEO. In this notebook, the raw data is normalized, and the phenotype data is cleaned and prepared. An expression-set object is created as the last step.</p>
<ul>
<li><a href="geodownload.html">Data download</a></li>
</ul>
<p>FOXO3 gene sets were obtained from 4 sources. One gene set comprises the top upregulated genes from a human erythroblast cell line treated with 50 nM 4-hydroxytamoxifen to stimulate FOXO3 activation (Bakker et al). Another comprises the top upregulated genes in a human colon cancer cell line treated with overexpression plasmids with FOXO3 compared to control (Delpeuch et al). A third is the top upregulated genes in wild type mice compared to FOXO1/FOXO3 double knock-out mice (wt_vs_FOXO1_FOXO3_DKO). Finally, a predicted FOXO3 target list based on comptational motif analysis was considered.</p>
</div>
<div id="ptcl-versus-control" class="section level2">
<h2>PTCL versus control</h2>
<p>We will examine the relative expression of FOXO3 targets in individual samples by disease group using GSVA. In the GSVA notebook, the expression data is filtered for protein coding genes only, and the probes are collapsed to gene level. Four gene sets containing different FOXO targets are used to investigate patterns of expression among different disease groups. The makeup of the gene sets was investigated by pathway enrichment. The differential gene expression was analyzed in PTCL versus control. These analysis make up figure 1 in the manuscript draft.</p>
<ul>
<li><a href="dge_analysis.html">GSVA</a></li>
<li><a href="pathway_enrichment.html">Which terms are enriched in the FOXO3 target gene sets?</a></li>
<li><a href="dge_analysis.html">Differential gene expression - PTCL vs control</a></li>
<li><a href="fgsea.html">Gene set enricment - PTCL vs control</a></li>
</ul>
</div>
<div id="foxo3-active-ptcl-versus-foxo3-non-active-ptcl" class="section level2">
<h2>FOXO3 active PTCL versus FOXO3 non-active PTCL</h2>
<p>Samples with a positive GSVA score in experimentally-derived FOXO3 target sets (indicating increased gene set expression compared to median gene expression) were termed “FOXO3 active” (positive score in all three of Delpeuch, Bakker, and wild-type vs FOXO1/FOXO3 DKO mouse). GSE19069 and GSE6338 datasets both contained FOXO3 active and inactive PTCL samples. PCA of gene expression showed significant batch effect between GSE19069 and GSE6338. GSE19069 has a larger number of FOXO3 active cases, and was used as the sole dataset for the analyses in this section. Coregulartory networks were assessed using ICA decomposition of the expression of the 1000 most variant genes. The differential gene expression</p>
<ul>
<li><a href="correlation.html">PCA and ICA</a></li>
<li><a href="foxo_ptcl_dge.html">Differential gene expression - FOXO3 active vs FOXO3 non-active</a></li>
</ul>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
